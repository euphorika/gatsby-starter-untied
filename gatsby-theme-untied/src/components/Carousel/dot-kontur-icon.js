import React from 'react'
import PropTypes from 'prop-types'

const DotIcon = ({ color }) => (
  <span
    className="dot-icon"
    dangerouslySetInnerHTML={{
      __html: `<svg data-name="Ebene 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 56.71 56.72">
  <path d="M28.35 56.72a28.36 28.36 0 1 1 28.36-28.36 28.38 28.38 0 0 1-28.36 28.36zm0-51.23a22.87 22.87 0 1 0 22.87 22.87A22.9 22.9 0 0 0 28.35 5.49z" fill="#3c3c3b"/>
</svg>
  `,
    }}
  />
)

DotIcon.propTypes = {
  color: PropTypes.string,
}

DotIcon.defaultProps = {
  color: '#3c3c3c',
}

export default DotIcon
