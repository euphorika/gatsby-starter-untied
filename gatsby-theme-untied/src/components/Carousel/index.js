import React from 'react'
import PropTypes from 'prop-types'

import styles from './styles.module.styl'

const Carousel = ({ children }) => {
  return (
    <div className={styles.Carousel}>
      <div className={styles.carouselSlides}>{children}</div>
    </div>
  )
}

Carousel.propTypes = {
  children: PropTypes.node.isRequired,
}

export default Carousel
