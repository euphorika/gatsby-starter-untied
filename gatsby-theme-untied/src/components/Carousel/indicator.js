import React from 'react'
import DotKontorIcon from './dot-kontur-icon.js'

class CarouselIndicator extends React.Component {
  render() {
    return (
      <li>
        <a
          href={this.props.href}
          className={
            this.props.index === this.props.activeIndex
              ? 'carousel-indicator'
              : null
          }
        >
          <DotKontorIcon />
        </a>
      </li>
    )
  }
}
export default CarouselIndicator
