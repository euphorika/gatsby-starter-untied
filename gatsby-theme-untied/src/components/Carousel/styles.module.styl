@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

.Carousel
  overflow: hidden
  margin-bottom: 50px
  margin: 0 auto
  width: 74%

  +above('m')
    padding: 0
    overflow: auto
    width: unset

  .carousel-slides
    display: flex
    overflow-x: auto
    scroll-snap-type: x mandatory
    scroll-behavior: smooth
    -webkit-overflow-scrolling: touch


  .carousel-slides::-webkit-scrollbar
    display: none

  .carousel-slides::-webkit-scrollbar-thumb
    display: none

  .carousel-slides::-webkit-scrollbar-track
    display: none

  .carousel-slides > section
    scroll-snap-align: start
    flex-shrink: 0
    height: 100%
    margin-right: 50px
    -webkit-transform-origin: center center
    transform-origin: center center
    -webkit-transform: scale(1)
    transform: scale(1)
    transition: -webkit-transform 0.5s
    transition: transform 0.5s
    transition: transform 0.5s, -webkit-transform 0.5s
    position: relative
    display: flex
    flex-direction: column
    justify-content: center
    +above('m')
      flex-shrink: unset
      margin-right: 15px


  .carousel-indicators
    display: flex
    justify-content: center
    
    +above('m')
      display: none

    li
      width: 20px
      height: 18px
      border: 0px solid primary-color
      border-radius: 50%
      margin-right: 10px
      text-align: center
      &:active
        background: primary-color

        .carousel-indicator
          width: 20px
          height: 20px
          border: 0px solid primary-color
