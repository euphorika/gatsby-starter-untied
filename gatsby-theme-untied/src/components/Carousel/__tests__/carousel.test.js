import React from 'react'
import renderer from 'react-test-renderer'
import Carousel from '../'

describe('Carousel', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Carousel>
          <div
            className="carouselSlides"
            id="slide0"
            id="slide1"
            id="slide2"
          />

        </Carousel>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
