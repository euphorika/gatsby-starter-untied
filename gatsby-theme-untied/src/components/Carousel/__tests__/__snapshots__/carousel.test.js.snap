// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel renders correctly 1`] = `
<div
  className="Carousel"
>
  <div
    className="carouselSlides"
    id="slideundefined"
  >
    <div
      className="carouselSlides"
      id="slide2"
    />
  </div>
  <div
    className="carouselDots"
  >
    <ul
      className="carouselIndicators"
    >
      <li>
        <a
          className={null}
          href="#slide-0"
        >
          <span
            className="dot-icon"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<svg data-name=\\"Ebene 1\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 56.71 56.72\\">
  <path d=\\"M28.35 56.72a28.36 28.36 0 1 1 28.36-28.36 28.38 28.38 0 0 1-28.36 28.36zm0-51.23a22.87 22.87 0 1 0 22.87 22.87A22.9 22.9 0 0 0 28.35 5.49z\\" fill=\\"#3c3c3b\\"/>
</svg>
  ",
              }
            }
          />
        </a>
      </li>
    </ul>
  </div>
</div>
`;
